### DISCUSSION TITLE
[ZK Stack] - How to upgrade ZKStack to be EIP4844 compatible

### DISCUSSION THREAD

**Discussion by Archethect** [zkSync-Community-Hub/zksync-developers#415] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

We are currently running our own ZKStack chain on testnet (connected to Sepolia).
What steps do we need to take to upgrade our chain to be EIP4844 compatible?
Which contracts do we have to upgrade? Are there any scripts for this? 

### Expected Behavior

The ZKStack chain would be EIP4844 compatible

### Current Behavior

The ZKStack chain is not 4844 compatible

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by zk-Lumi** [comment, marked as answer]
Hi @Archethect,

We working on a formal process for hyperchain upgrades. We really want to make sure this process is careful, safe, and easy for all hyperchains. This ties with some other efforts we are working on atm, so I don't have a clear ETA for this yet besides some time in April (there are a lot of moving pieces across teams that affect this).

If it makes sense to from your side, might be worth redeploying now with a new deployment, and will make sure to have formal progress for next upgrades.

**Reply by Archethect** [reply]
Ok, makes sense. We need to redeploy anyway for ERC20 gas token support, so we will combine both features by the deployment of a new test chain.

**Reply by SuccinctPaul** [reply]
hi @zk-Lumi, Any updates now for how to upgrade the zkstack?
